### ISSUE TITLE
Paper: Why Gradients Rapidly Increase Near the End of Training

### ISSUE THREAD

**Issue by rockerBOO** [LoganBooker/prodigy-plus-schedule-free#30]
Possibly a small change to weight decay to make it a little more stable. 

https://arxiv.org/abs/2506.02285 

![Image](https://github.com/user-attachments/assets/ce4dc8d8-a7e9-4144-95c8-9b19ff6a750f)

**Comment by LoganBooker** [comment]
This is actually very similar to how decoupled weight decay is meant to be implemented, that is, as a ratio of `lr / max_lr` (https://optimi.benjaminwarner.dev/fully_decoupled_weight_decay/), where this is `(lr ** 2) / max_lr` instead. Which makes me wonder if you could just apply it to all layers, rather than just normalised ones. The paper doesn't seem to have conducted tests on all layers, which would have been good to see for the sake of comparison.
